core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java

```diff
@@ -239,13 +241,18 @@ public static void reportException(@NonNull String name, @NonNull Throwable e) {
             // We call the methods in this order because if the missing class is not java related, we don't loop over the
             // stack trace to look if it's not thrown from an ignored place avoiding an impact on performance.
             if (isFromMovedPackage(name) && !calledFromIgnoredPlace(e)) {
-                events.put(name, e);
-                if (LOGGER.isLoggable(Level.WARNING))
+                if (LOGGER.isLoggable(Level.WARNING) && !wasAlreadyReported(name)) {
                     LOGGER.log(Level.WARNING, "Added a missed class for missing class telemetry. Class: " + name, e);
+                }
+                events.put(name, e);
```

Comment:
Should it be added even if already there?

Reply:
The put method is not the Map one, is a custom method which makes two things:

- Put the event in the list of already printed, if needed
- Put the event in the list of events to send to Telemetry, if there is space for it (only 100 per day)